nutils/types.py

```diff
@@ -1475,16 +1496,22 @@ def _parse(cls, s):
     return value, powers
 
   def __new__(cls, s):
-    v, powers = cls._parse(s)
-    if not hasattr(cls, 'bound'):
-      cls = type('{}:{}'.format(cls.__name__, powers), (cls,), dict(bound=powers))
-    elif cls.bound != powers:
-      raise ValueError('invalid unit: expected {}, got {}'.format(cls.bound, powers))
-    self = float.__new__(cls, v)
-    self._str = s
-    return self
+    value, powers = cls._parse(s)
+    if not hasattr(cls, 'unit'):
+      cls = cls[s]
+    elif cls.powers != powers:
+      raise ValueError('invalid unit: expected {}, got {}'.format(cls.powers, powers))
+    return float.__new__(cls, value)
+
+  @classmethod
+  def __stringly_loads__(cls, s):
+    return cls(s)
+
+  @classmethod
+  def __stringly_dumps__(cls, v):
+    if not hasattr(cls, 'unit'):
+      raise Exception('only bound units can be dumped')
+    return '{:f}'.format(v/cls.value).strip('0').rstrip('.') + cls.unit
```

Comment:
Maybe you should add an `if not isinstance(v, float): raise Exception/stringly.error.SerializationError` here.